### PR TITLE
docs(plugins): Update plugin creators guide

### PR DESCRIPTION
**Pull request by aimeeu**
Update plugin creators guide for PF4J functionality added in 1.19

- Created folder for plugin creators guide since the single page was getting too long; this will also allow for more pages if we decide to walkthrough the other plugin examples
- Moved walkthrough of pf4jStagePlugin to new guide

 Because we'd like to get this merged for Gardening Days, I added "Help us improve this section by submitting a pull request!" to some sections.  If that's not OK, I can remove those sections and add them back later. Same applies to the Project Configuration page.

**Note**: merging this PR will break a link in https://blog.spinnaker.io/cant-go-outside-dip-your-toes-in-the-spinnaker-plugins-pool-1076f74003b

Co-authored by the Armory plugins team and Cameron Fieber (cfieber)

### CODE REVIEW

**Comment by aimeeu** [comment]
This PR builds, deploys, and renders without error locally. However, the official build fails on the `rake` job - it doesn't like the reference style links???

```
 <a href="/guides/developer/plugin-creators-guide/overview/">Plugin Creators Guide Overview</a>
- ./_site/guides/developer/plugin-creators/overview/index.html
  *  [https://github.com/spinnaker-plugin-examples/springExamplePlugin] is an invalid URL (line 2737)
     <a href="%5Bhttps://github.com/spinnaker-plugin-examples/springExamplePlugin%5D">springExamplePlugin</a>


  *  internally linking to EchoEventListener, which does not exist (line 2687)
     <a href="EchoEventListener">EventListener</a>
  *  internally linking to KorkSecretEngine, which does not exist (line 2715)
     <a href="KorkSecretEngine">SecretEngine</a>
  *  internally linking to OrcaPreconfiguredJobConfigurationProvider, which does not exist (line 2682)
     <a href="OrcaPreconfiguredJobConfigurationProvider">PreconfiguredJobConfigurationProvider</a>
  *  internally linking to OrcaSimpleStage, which does not exist (line 2681)
     <a href="OrcaSimpleStage">SimpleStage</a>
  *  internally linking to pf4jPluginWithoutExtensionPoint, which does not exist (line 2715)
     <a href="pf4jPluginWithoutExtensionPoint">README</a>
- ./_site/guides/user/plugin-users/index.html
  *  internally linking to /guides/developer/plugin-creators, which does not exist (line 2648)
     <a href="/guides/developer/plugin-creators">Plugin Creators Guide</a>
- ./_site/news/editions/edition2/index.html
  *  internally linking to /guides/developer/plugin-creators/, which does not exist (line 441)
     <a href="/guides/developer/plugin-creators/">Plugin Creators Guide</a>
rake aborted!
HTML-Proofer found 9 failures!
```

File: guides/developer/plugin-creators/extensionpoint-plugin-walkthrough.md
Comment (dorbin) [inline guides/developer/plugin-creators/extensionpoint-plugin-walkthrough.md]: Consider using the alpha include, like [here](https://github.com/spinnaker/spinnaker.github.io/blob/master/guides/user/managed-delivery/index.md). In that example, there's another paragraph referring to its alphaness, but I think we should be ok with just the include.

File: guides/developer/plugin-creators/extensionpoint-plugin-walkthrough.md
Comment (dorbin) [inline guides/developer/plugin-creators/extensionpoint-plugin-walkthrough.md]: I haven't used reference-style links, so I can't help you with those rake errors, but would ordinary in-line links work for your purposes?

File: guides/developer/plugin-creators/overview.md
Comment (dorbin) [inline guides/developer/plugin-creators/overview.md]: Is there a reason it has to be IntelliJ?

File: guides/developer/plugin-creators/overview.md
Comment (dorbin) [inline guides/developer/plugin-creators/overview.md:29]: "operator"
It's for developer's right? That is, developers of Spinnaker, not Spinnaker users. And operators are separate; they're the ones installing and running it. 

Is this an accurate version of what you're saying?:
"You can use a plugin to extend Spinnaker with custom functionality"

File: guides/developer/plugin-creators/overview.md
Comment (dorbin) [inline guides/developer/plugin-creators/overview.md]: And if the above comment is correct, then I think you could just delete that whole sentence and go with this one.

File: guides/developer/plugin-creators/overview.md
Comment (dorbin) [inline guides/developer/plugin-creators/overview.md:35]:
```suggestion
* It's the easiest.
  Use the `@Extension` annotation and implement the methods declared in your chosen extension point.
```

File: guides/developer/plugin-creators/overview.md
Comment (dorbin) [inline guides/developer/plugin-creators/overview.md:36]:
```suggestion
* Spinnaker loads the plugin in an isolated classpath.
```

File: guides/developer/plugin-creators/overview.md
Comment (dorbin) [inline guides/developer/plugin-creators/overview.md]:
```suggestion
* It has the least amount of maintenance work.
  Updates to Spinnaker are not likely to break your plugin.
```

File: guides/developer/plugin-creators/overview.md
Comment (dorbin) [inline guides/developer/plugin-creators/overview.md]: Link from here to the Slack space?

File: guides/developer/plugin-creators/overview.md
Comment (dorbin) [inline guides/developer/plugin-creators/overview.md:31]:
```suggestion
# `ExtensionPoint` plugins
```

Comment (dorbin) [reply guides/developer/plugin-creators/overview.md:31]: (Unless `ExtensonPoint` isn't actually an object in this context, in which case it might be better to just have "Extension point plugins"

File: guides/developer/plugin-creators/overview.md
Comment (dorbin) [inline guides/developer/plugin-creators/overview.md]: This is _slightly_ different from how you said it above. If there's no difference between the advantage here and the advange for extension point-style plugins, I'd say it exactly the same in both places.

File: guides/developer/plugin-creators/overview.md
Comment (dorbin) [inline guides/developer/plugin-creators/overview.md]: Is this (self?)-question to be resolved before this gets merged?

File: _data/navigation.yml
Comment (robzienert) [inline _data/navigation.yml]: This title doesn't make a lot of sense. Are they creating an Extension Point, or are they creating a Plugin that implements Extension Points?

File: guides/developer/plugin-creators/overview.md
Comment (robzienert) [reply guides/developer/plugin-creators/overview.md]: There's no reason that it needs to be IntelliJ, no.

File: guides/developer/plugin-creators/overview.md
Comment (robzienert) [reply guides/developer/plugin-creators/overview.md:29]: There are 3 personas in plugins:

1. Core Contributors: Those who create extension points in the core services.
2. Developers: Those who create extensions / plugins implementing the extension points offered by services.
3. Operators: Those who install plugins into their Spinnaker installations.

I think it's very, very important that each persona gets its own variation of docs for the plugin experience, since each persona is going to have to know different things when it comes to this larger feature.

File: guides/developer/plugin-creators/extensionpoint-plugin-walkthrough.md
Comment (robzienert) [inline guides/developer/plugin-creators/extensionpoint-plugin-walkthrough.md]: This wording is deceptive. Organizing these values into `gradle.properties` is optional, but these versions are very much mandatory.

File: guides/developer/plugin-creators/extensionpoint-plugin-walkthrough.md
Comment (robzienert) [inline guides/developer/plugin-creators/extensionpoint-plugin-walkthrough.md]: > These dependencies must be loaded from the service's classloader, not the plugin classloader.

This seems like something that would be a better fit for internals docs, rather than a getting started doc. The way it's worded here I think would lead to confusion: To me, it reads like It's hinting at an action that developers must take, but there really isn't any action to take on their part.

File: guides/developer/plugin-creators/extensionpoint-plugin-walkthrough.md
Comment (robzienert) [inline guides/developer/plugin-creators/extensionpoint-plugin-walkthrough.md]:
```suggestion
Define Spinnaker Stages with IStageTypeConfig. Required [options](https://github.com/spinnaker/deck/blob/abac63ce5c88b809fcf5ed1509136fe96489a051/app/scripts/modules/core/src/domain/IStageTypeConfig.ts):
```

File: guides/developer/plugin-creators/extensionpoint-plugin-walkthrough.md
Comment (robzienert) [inline guides/developer/plugin-creators/extensionpoint-plugin-walkthrough.md]:
```suggestion
- `label` -> The name of the Stage
- `description` -> Long form that describes what the Stage actually does
- `key` -> A unique name for the Stage
- `component` -> The rendered React component
```

File: guides/developer/plugin-creators/extensionpoint-plugin-walkthrough.md
Comment (robzienert) [inline guides/developer/plugin-creators/extensionpoint-plugin-walkthrough.md]: Is this a Halyard config format? I've not seen anything that resembles this before.

File: guides/developer/plugin-creators/overview.md
Comment (robzienert) [inline guides/developer/plugin-creators/overview.md:27]: We need a taxonomy section. This PR is mixing terminology up and it'll confuse people.

* Extension Point: Defined by a Spinnaker service, it represents an official contract for which extensions can be written against. Written by Core Contributors.
* Extension: A piece of code that implements an Extension Point. Written by Developers.
* Plugin: A collection of Extensions that can be bundled together and shipped as a single artfiact. Also written by Developers.

File: guides/developer/plugin-creators/overview.md
Comment (robzienert) [reply guides/developer/plugin-creators/overview.md:31]: `ExtensionPoint` is a real thing, yes.

File: guides/developer/plugin-creators/overview.md
Comment (aimeeu) [reply guides/developer/plugin-creators/overview.md]: Moved out of overview and into Pipeline Stage Plugin Walkthrough.  The IDE does need to be IntelliJ for the walkthrough, which will how to use  IntelliJ to debug the plugin.

File: guides/developer/plugin-creators/overview.md
Comment (aimeeu) [reply guides/developer/plugin-creators/overview.md:29]: @robzienert thanks for the great feedback. There already is a [Plugin Users Guide](https://www.spinnaker.io/guides/user/plugin-users/), which needs to be updated to state Spinnaker 1.19.4 and Halyard 1.34.

I've created a GitHub [issue](https://github.com/spinnaker/spinnaker.github.io/issues/1776) to create an Extension Point Creators Guide.

File: guides/developer/plugin-creators/extensionpoint-plugin-walkthrough.md
Comment (aimeeu) [reply guides/developer/plugin-creators/extensionpoint-plugin-walkthrough.md]: removed this and added as code comments  in the pf4jStagePlugin repo

File: guides/developer/plugin-creators/extensionpoint-plugin-walkthrough.md
Comment (aimeeu) [reply guides/developer/plugin-creators/extensionpoint-plugin-walkthrough.md]: Cameron Fieber wrote the project configuration stuff. I changed the wording to what you suggest.

File: guides/developer/plugin-creators/extensionpoint-plugin-walkthrough.md
Comment (aimeeu) [reply guides/developer/plugin-creators/extensionpoint-plugin-walkthrough.md]: changed "must be" to "are"

File: guides/developer/plugin-creators/extensionpoint-plugin-walkthrough.md
Comment (aimeeu) [reply guides/developer/plugin-creators/extensionpoint-plugin-walkthrough.md]: removed this and added as a code comments in the pf4jStagePlugin repo

File: guides/developer/plugin-creators/extensionpoint-plugin-walkthrough.md
Comment (aimeeu) [reply guides/developer/plugin-creators/extensionpoint-plugin-walkthrough.md]: Don't know where Brandon go this - removed along with the entire "frontend" section, which was outdated.

File: guides/developer/plugin-creators/overview.md
Comment (aimeeu) [reply guides/developer/plugin-creators/overview.md:27]: added

**Comment by aimeeu** [comment]
@dorbin @robzienert @brian-armory 

Because we'd like to get this merged for Gardening Days, I added "Help us improve this section by submitting a pull request!" to some sections. If that's not OK, I can remove those sections and add them back later. Same applies to the Project Configuration page.

Note: merging this PR will break a link in https://blog.spinnaker.io/cant-go-outside-dip-your-toes-in-the-spinnaker-plugins-pool-1076f74003b

**Review by dorbin — APPROVED** [review]
In the interest of timeliness, I'm approving. 
There are some merge conflicts though.

**Comment by aimeeu** [comment]
I ran into rebase hell, so I'm closing this PR in favor of #1784

**Pull request closed without merging**
